source/extensions/filters/network/mysql_proxy/mysql_utils.cc

```diff
@@ -216,6 +220,147 @@ DecodeStatus BufferHelper::peekHdr(Buffer::Instance& buffer, uint32_t& len, uint
   return DecodeStatus::Success;
 }
 
+AuthMethod AuthHelper::authMethod(uint32_t cap, const std::string& auth_plugin_name) {
+  if (auth_plugin_name.empty()) {
+    /*
+     * https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_connection_phase.html
+     */
+    bool v41 = cap & CLIENT_PROTOCOL_41;
+    bool sconn = cap & CLIENT_SECURE_CONNECTION;
+    if (!v41 || !sconn) {
+      return AuthMethod::OldPassword;
+    }
+    return AuthMethod::NativePassword;
+  }
+
+  if (auth_plugin_name == "mysql_old_password") {
```

Comment:
Can you define constants for strings used here and in authPluginName? Like in https://github.com/envoyproxy/envoy/blob/main/source/extensions/filters/network/postgres_proxy/postgres_decoder.cc#L17

Reply:
OK.